server/src/test/java/org/solarbank/server/integration/EnergySavingControllerIT.java

```diff
@@ -86,4 +86,13 @@ public void energySavingController_throwException_status500() throws Exception {
             .andExpect(jsonPath("$.Error.Status").value(HttpStatus.INTERNAL_SERVER_ERROR.getReasonPhrase()))
             .andExpect(jsonPath("$.Error.Message").value(ErrorMessage.INTERNAL_SERVER_ERROR.getMessage()));
     }
+
+    @Test
+    public void nonExistentPath_postRequest_returnsNotFound404() throws Exception  {
```

Comment:
I think we can adjust all of the tests in this file so that we don't throw an exception from the test.

We can do this by wrapping our test in a `try-catch` block and failing the test if an error is thrown using `import static org.junit.jupiter.api.Assertions.fail`.

for example the first test in the file could be written like:

```
    @Test
    public void energySavingController_validRequest_status200() {
        CalculateRequest calculateRequest = createCalculateRequest();

        try {
            String validRequest = mapToString(calculateRequest);

            mockMvc.perform(post("/v1.0/api/calculate")
                    .contentType(MediaType.APPLICATION_JSON)
                    .content(validRequest))
                .andExpect(status().isOk())
                .andDo(print())
                .andExpect(jsonPath("$.EnergyGenPerYear").value(1.0))
                .andExpect(jsonPath("$.EnergyGenPerMonth.January").value(0.1))
                .andExpect(jsonPath("$.EnergyGenPerMonth.February").value(0.2))
                .andExpect(jsonPath("$.SavingsPerYear.CurrencyCode").value("USD"))
                .andExpect(jsonPath("$.SavingsPerYear.Amount").value(1000.0));
        } catch (Exception e) {
           fail("failed to perform api requests as expected");
        }
    }
```

To me this is cleaner as you don't need to have the `throws Exception` on every method signature.